src/test/lib/optimizer/strategy/expression_reduction_rule_test.cpp

```diff
@@ -162,6 +162,15 @@ TEST_F(ExpressionReductionRuleTest, RewriteLikePrefixWildcard) {
   auto expression_m = std::shared_ptr<AbstractExpression>(or_(a, like_(s, "RED")));
   ExpressionReductionRule::rewrite_like_prefix_wildcard(expression_m);
   EXPECT_EQ(*expression_m, *or_(a2, equals_(s, "RED")));
+
+  // Case-insensitive patterns should not be rewritten.
+  auto expression_n = std::shared_ptr<AbstractExpression>(ilike_(s, "RED%"));
+  ExpressionReductionRule::rewrite_like_prefix_wildcard(expression_n);
```

Comment:
I haven't seen any changes to the rule. Add a comment to explicitly tell that we don't do certain tricks for ilike?